.gitlab-ci.yml

```diff
@@ -268,6 +268,16 @@ cargo-fmt:
     - cargo +nightly fmt --all -- --check
   allow_failure:                   true
 
+cargo-clippy:
+  stage:                           test
+  <<:                              *docker-env
+  <<:                              *test-refs
+  script:
+    - rustup +nightly component add clippy
+    - cargo +nightly install cargo-lints
+    - cargo +nightly lints clippy
+  allow_failure:                   true
```

Comment:
if it's green and not flaky, it should not be allowed to fail. Otherwise, no one will pay attention to it.

Reply:
Also, the job takes up 7 minutes, this is quite a lot. I wonder if it can be merged with some cache-wise similar job to optimize our concurrency queue in pipelines.

Reply:
95% of that is just redoing work that other jobs will have done so if we tagged it onto check it would be a lot cheaper.

Reply:
While we could tack it on to another job, initially it would be better kept separate as we want to allow failures while everyone is getting used to clippy. (In the same way, cargo fmt was not mandatory initially. Once it's a mandatory job it becomes easier to run it after another job.)